gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -457,4 +457,9 @@ private Collection<KnoxToken> fetchTokens(String userName, boolean createdBy) {
     });
     return tokens;
   }
+
+  @Override
+  public boolean isMigrationTarget() {
```

Comment:
Instead of a method to override in every impl, you could have those implementations that support migration implement a MigrationTarget interface.

Reply:
Since the `DefaultTokenStateService` implements that method already, there is no need to implement it everywhere. The only override of this method is needed when you want to make that implementation a migration target. That's what I did in `JDBCTokenStateService`.
From this point of view, making that class implement an interface or override this method, doesn't make any difference, therefore I don't mind changing to the interface approach.

Reply:
Done.